src/main/java/games/strategy/triplea/settings/ClientSettingUiBinding.java

```diff
@@ -0,0 +1,213 @@
+          + "listing file, update it, and put the path to that file here."),
+  TEST_LOBBY_HOST_BINDING(
+      "Lobby Host Override",
+      SettingType.TESTING,
+      SelectionComponentFactory.textField(ClientSetting.TEST_LOBBY_HOST),
+      "Overrides the IP address or hostname used to connect to the lobby. Useful for connecting to a test lobby."),
+  TEST_LOBBY_PORT_BINDING(
+      "Lobby Port Override",
+      SettingType.TESTING,
+      SelectionComponentFactory.intValueRange(ClientSetting.TEST_LOBBY_PORT, 1, 99999),
+      "Specifies the port for connecting to a test lobby."),
+  TRIPLEA_FIRST_TIME_THIS_VERSION_PROPERTY_BINDING(
+      "Show First Time Prompts",
+      SettingType.GAME,
+      ClientSetting.TRIPLEA_FIRST_TIME_THIS_VERSION_PROPERTY,
+      "Setting to true will trigger for any first time prompts to be shown"),
+  SAVE_GAMES_FOLDER_PATH_BINDING(
+      "Saved Games Folder",
+      SettingType.FOLDER_LOCATIONS,
+      SelectionComponentFactory.folderPath(ClientSetting.SAVE_GAMES_FOLDER_PATH),
+      "The folder where saved game files will be stored by default"),
+  USER_MAPS_FOLDER_PATH_BINDING(
+      "Maps Folder",
+      SettingType.FOLDER_LOCATIONS,
+      SelectionComponentFactory.folderPath(ClientSetting.USER_MAPS_FOLDER_PATH),
+      "The folder where game engine will download and find map files."),
+  WHEEL_SCROLL_AMOUNT_BINDING(
+      "Mouse Wheel Scroll Speed",
+      SettingType.MAP_SCROLLING,
+      SelectionComponentFactory.intValueRange(ClientSetting.WHEEL_SCROLL_AMOUNT, 10, 300),
+      "How fast the map will scroll when using the mouse wheel"),
+  PROXY_CHOICE(
+      "Network Proxy",
+      SettingType.NETWORK_PROXY,
+      SelectionComponentFactory.proxySettings(),
+      "Configure TripleA's Network and Proxy Settings\n"
+      + "This only effects Play-By-Forum games, dice servers, and map downloads.");
+
+
+  final SettingType type;
+  final String title;
+  final String description;
+  private final Supplier<SelectionComponent> selectionComponentBuilder;
+
+  private SelectionComponent selectionComponent;
+
+  ClientSettingUiBinding(
+      final String title,
+      final SettingType type,
+      final Supplier<SelectionComponent> selectionComponent,
+      final String description) {
+    this.title = Preconditions.checkNotNull(Strings.emptyToNull(title));
+    this.type = Preconditions.checkNotNull(type);
+    this.selectionComponentBuilder = selectionComponent;
+    this.description = Preconditions.checkNotNull(Strings.emptyToNull(description));
+  }
+
+  ClientSettingUiBinding(
+      final String title,
+      final SettingType type,
+      final ClientSetting clientSetting,
+      final String description) {
+    this(title, type, () -> SelectionComponentFactory.booleanRadioButtons(clientSetting), description);
+  }
+
+  @Override
+  public JComponent buildSelectionComponent() {
+    return current().getJComponent();
+  }
+
+  private synchronized SelectionComponent current() {
```

Comment:
Similar to my previous comment about the need for `synchronized` on `SettingsWindow#open()`...  This code isn't quite the same because `ClientSettingUiBinding` is not a Swing component.  However, the `Ui` in its name suggests to me that it is only called by UI code, and thus is effectively single threaded.  Is that not true?

If not, and the synchronization is required, then you might also want to synchronize `dispose()`.  While assignment to an object reference is atomic, the publication of the updated value to other threads is not guaranteed without some kind of memory barrier (unless the field happens to be `volatile`).